pkg/secretsencrypt/controller.go

```diff
@@ -1,246 +0,0 @@
-	}
-	if err := cluster.Save(h.ctx, h.controlConfig, true); err != nil {
-		h.recorder.Event(nodeRef, corev1.EventTypeWarning, secretsUpdateErrorEvent, err.Error())
-		return node, err
-	}
-	return node, nil
-}
-
-// validateReencryptStage ensures that the request for reencryption is valid and
-// that there is only one active reencryption at a time
-func (h *handler) validateReencryptStage(node *corev1.Node, annotation string) (bool, error) {
-	split := strings.Split(annotation, "-")
-	if len(split) != 2 {
-		err := fmt.Errorf("invalid annotation %s found on node %s", annotation, node.ObjectMeta.Name)
-		return false, err
-	}
-	stage := split[0]
-	hash := split[1]
-
-	// Validate the specific stage and the request via sha256 hash
-	if stage != EncryptionReencryptRequest {
-		return false, nil
-	}
-	if reencryptRequestHash, err := GenReencryptHash(h.controlConfig.Runtime, EncryptionReencryptRequest); err != nil {
-		return false, err
-	} else if reencryptRequestHash != hash {
-		err = fmt.Errorf("invalid hash: %s found on node %s", hash, node.ObjectMeta.Name)
-		return false, err
-	}
-	reencryptActiveHash, err := GenReencryptHash(h.controlConfig.Runtime, EncryptionReencryptActive)
-	if err != nil {
-		return false, err
-	}
-	labelSelector := labels.Set{util.ControlPlaneRoleLabelKey: "true"}.String()
-	nodes, err := h.nodes.List(metav1.ListOptions{LabelSelector: labelSelector})
-	if err != nil {
-		return false, err
-	}
-	for _, node := range nodes.Items {
-		if ann, ok := node.Annotations[EncryptionHashAnnotation]; ok {
-			split := strings.Split(ann, "-")
-			if len(split) != 2 {
-				return false, fmt.Errorf("invalid annotation %s found on node %s", ann, node.ObjectMeta.Name)
-			}
-			stage := split[0]
-			hash := split[1]
-			if stage == EncryptionReencryptActive && hash == reencryptActiveHash {
-				return false, fmt.Errorf("another reencrypt is already active")
-			}
-		}
-	}
-	return true, nil
-}
-
-func (h *handler) updateSecrets(nodeRef *corev1.ObjectReference) error {
-	secretPager := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
-		return h.k8s.CoreV1().Secrets(metav1.NamespaceAll).List(h.ctx, opts)
-	}))
-	secretPager.PageSize = secretListPageSize
-
-	i := 0
-	if err := secretPager.EachListItem(h.ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
-		secret, ok := obj.(*corev1.Secret)
-		if !ok {
-			return errors.New("failed to convert object to Secret")
-		}
-		if _, err := h.k8s.CoreV1().Secrets(secret.Namespace).Update(h.ctx, secret, metav1.UpdateOptions{}); err != nil && !apierrors.IsConflict(err) {
-			return fmt.Errorf("failed to update secret: %v", err)
-		}
-		if i != 0 && i%50 == 0 {
-			h.recorder.Eventf(nodeRef, corev1.EventTypeNormal, secretsProgressEvent, "reencrypted %d secrets", i)
-		}
-		i++
-		return nil
-	}); err != nil {
-		return err
-	}
-
-	h.recorder.Eventf(nodeRef, corev1.EventTypeNormal, secretsUpdateCompleteEvent, "completed reencrypt of %d secrets", i)
```

Comment:
I know it runs real fast now, but do we maybe still want to emit some of these events in case of error or success? Just in case anyone was expecting to see them when they rotate?

Reply:
We do still have logging, its just been converting to Logrus Info events
https://github.com/k3s-io/k3s/pull/10612/files#diff-9437ed627f73b3d14166963e7aba13ac15e8da8311ab750072c2b1c4a4742145R421
https://github.com/k3s-io/k3s/pull/10612/files#diff-9437ed627f73b3d14166963e7aba13ac15e8da8311ab750072c2b1c4a4742145R429

Reply:
right but those only show up in the server logs, the events that we used to emit show up in the Rancher UI and in `kubectl describe node`.